components/script/dom/characterdata.rs

```diff
@@ -91,12 +92,9 @@ impl CharacterDataMethods for CharacterData {
 
     // https://dom.spec.whatwg.org/#dom-characterdata-data
     fn SetData(&self, data: DOMString) {
-        let old_length = self.Length();
-        let new_length = data.encode_utf16().count() as u32;
-        *self.data.borrow_mut() = data;
-        self.content_changed();
-        let node = self.upcast::<Node>();
-        node.ranges().replace_code_units(node, 0, old_length, new_length);
+        // Reusing ReplaceData to ensure we will also trigger
+        // mutation observers.
+        self.ReplaceData(0, self.Length(), data).unwrap_or(());
```

Comment:
A bit unfortunate, this is probably a tad slower...

We had `content_changed`, which we can probably inline now in ReplaceData... But maybe it's worth to do both `content_will_change` and `content_changed`, and keep the fast path here, triggering the mutation in `content_will_change`.

Reply:
I added a `queue_mutation_record` method that is called from both `SetData` and `ReplaceData`.